oktapam/client/cloud_connection.go

```diff
@@ -0,0 +1,199 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"regexp"
+
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/logging"
+	"github.com/okta/terraform-provider-oktapam/oktapam/utils"
+	"github.com/tomnomnom/linkheader"
+)
+
+type CloudConnection struct {
+	ID                     *string                 `json:"id,omitempty"`
+	Name                   *string                 `json:"name"`
+	Provider               *string                 `json:"provider"`
+	CloudConnectionDetails *CloudConnectionDetails `json:"cloud_connection_details"`
+}
+
+type CloudConnectionDetails struct {
+	AccountId  *string `json:"account_id"`
+	ExternalId *string `json:"external_id"`
+	RoleArn    *string `json:"role_arn"`
+}
+
+type CloudConnectionsListResponse struct {
+	CloudConnections []CloudConnection `json:"list"`
+}
+
+func (c CloudConnection) Exists() bool {
+	return utils.IsNonEmpty(c.ID)
+}
+
+func (c CloudConnection) ToResourceMap() map[string]any {
+	m := make(map[string]any)
+
+	if c.ID != nil {
+		m[attributes.ID] = *c.ID
+	}
+	if c.Name != nil {
+		m[attributes.Name] = *c.Name
+	}
+	if c.Provider != nil {
+		m[attributes.CloudConnectionProvider] = c.Provider
+	}
+	if c.CloudConnectionDetails != nil {
+		flattenedDetails := make([]any, 1)
+		flattenedDetail := make(map[string]any)
+		if c.CloudConnectionDetails.AccountId != nil {
+			flattenedDetail[attributes.CloudConnectionAccountId] = *c.CloudConnectionDetails.AccountId
+		}
+		if c.CloudConnectionDetails.ExternalId != nil {
+			flattenedDetail[attributes.CloudConnectionExternalId] = *c.CloudConnectionDetails.ExternalId
+		}
+		if c.CloudConnectionDetails.RoleArn != nil {
+			flattenedDetail[attributes.CloudConnectionRoleARN] = *c.CloudConnectionDetails.RoleArn
+		}
+		flattenedDetails[0] = flattenedDetail
+		m[attributes.CloudConnectionDetails] = flattenedDetails
+	}
+
+	return m
+}
+
+func isCloudConnectionDataValid(cloudConnection CloudConnection) bool {
+	if cloudConnection.Name == nil || cloudConnection.CloudConnectionDetails == nil || cloudConnection.CloudConnectionDetails.AccountId == nil || cloudConnection.CloudConnectionDetails.ExternalId == nil || cloudConnection.Provider == nil || cloudConnection.CloudConnectionDetails.RoleArn == nil {
+		return false
+	}
+
+	const PROVIDER = "aws"
```

Comment:
sorry, but can we move this to the top of the file?
also nit: `AWS_PROVIDER_NAME = "aws"`


same for `namePattern := regexp.MustCompile(`^[A-Za-z0-9-_.]+$`)`

That is what i meant in my previous comment.
those values that are fixed should be defined at the top

Reply:
I renamed the variable to `AWS_PROVIDER_NAME`.

But I don't think it's a good idea to move those variables (the provider and the 3 other regexp) to the top of the file because they will exposed as package-level variables, which I don't think is a good idea because those variables should be scooped to this function only, they're not really that useful to other areas of the code.

Reply:
i am not sure,if that is a big deal for now
but if we wanted to use the regex across the package then it would be better to be declared there.

anyway i am okay with this too!